internal/services/synapse/synapse_integration_runtime_self_hosted_resource.go

```diff
@@ -0,0 +1,187 @@
+package synapse
+
+import (
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/synapse/mgmt/2021-03-01/synapse"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/tf"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/synapse/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/synapse/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+func resourceSynapseIntegrationRuntimeSelfHosted() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceSynapseIntegrationRuntimeSelfHostedCreateUpdate,
+		Read:   resourceSynapseIntegrationRuntimeSelfHostedRead,
+		Update: resourceSynapseIntegrationRuntimeSelfHostedCreateUpdate,
+		Delete: resourceSynapseIntegrationRuntimeSelfHostedDelete,
+
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := parse.IntegrationRuntimeID(id)
+			return err
+		}),
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*pluginsdk.Schema{
+			"name": {
+				Type:     pluginsdk.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringMatch(
+					regexp.MustCompile(`^[A-Za-z0-9]+(?:-[A-Za-z0-9]+)*$`),
+					`Invalid name for Self-Hosted Integration Runtime: minimum 3 characters, must start and end with a number or a letter, may only consist of letters, numbers and dashes and no consecutive dashes.`,
+				),
+			},
+
+			"synapse_workspace_id": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.WorkspaceID,
+			},
+
+			"description": {
+				Type:         pluginsdk.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"auth_key_1": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+
+			"auth_key_2": {
```

Comment:
should we make this a list ?

Reply:
@katbyte , I think it's not necessary,  data_factory_integration_runtime_self_hosted is a very similar resource and it uses 2 properties, we can keep consistent with it.

https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/data_factory_integration_runtime_self_hosted#auth_key_1

Reply:
ok so looking at the docs & given 1 & 2 are not descriptive, and it seems key 1 is primary, and 2 is secondary can we make this authorization 
```suggestion
			"authorization_key_primary": {
				Type:     pluginsdk.TypeString,
				Computed: true,
			},

			"authorization_key_secondary": {
```

and lets remove the abbreviation

datafactory should also be updated as 1 & 2 do no reflect what the keys are, could we do that/add a todo 3.0 there?